src/Http/Response.php

```diff
@@ -2056,13 +2061,45 @@ public function getCookies()
     {
         $out = [];
         foreach ($this->_cookies as $cookie) {
-            $out[$cookie->getName()] = $cookie->toArrayResponse();
+            $out[$cookie->getName()] = $this->toArrayResponse($cookie);
         }
 
         return $out;
     }
 
     /**
+     * Convert the cookie into an array of its properties.
+     *
+     * This method is compatible with the historical behavior of Cake\Http\Response,
+     * where `httponly` is `httpOnly` and `expires` is `expire`
+     *
+     * @param \Cake\Http\Cookie\CookieInterface $cookie Cookie object.
+     * @return array
+     */
+    protected function toArrayResponse(CookieInterface $cookie)
+    {
+        if ($cookie instanceof Cookie) {
+            return $cookie->toArrayResponse();
+        }
```

Comment:
Same as above. It looks like the compat methods could be removed from the `Cookie` object and put here instead.